ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/group/GroupListCommand.java

```diff
@@ -51,15 +53,25 @@ public String getCommandName() {
   @Override
   public int run(CommandLine cl) throws IOException {
     super.run(cl);
-    String strAddr = cl.getOptionValue(SERVER_ADDRESS_OPTION_NAME);
-    final InetSocketAddress serverAddress = parseInetSocketAddress(strAddr);
-    final RaftPeerId peerId = RaftUtils.getPeerId(serverAddress);
+    RaftPeerId peerId = null;
+    String strAddr = null;
+    if(cl.hasOption(SERVER_ADDRESS_OPTION_NAME)) {
+      strAddr = cl.getOptionValue(SERVER_ADDRESS_OPTION_NAME);
+      final InetSocketAddress serverAddress = parseInetSocketAddress(strAddr);
+      peerId = RaftUtils.getPeerId(serverAddress);
+    } else if (cl.hasOption(PEER_ID_OPTION_NAME)) {
+      peerId = RaftPeerId.getRaftPeerId(cl.getOptionValue(PEER_ID_OPTION_NAME));
+      strAddr = getRaftGroup().getPeer(peerId).getAddress();
+    }
```

Comment:
Add an else-clause to throw IllegalArgumentException.

Reply:
@szetszwo Thanks for your review!
Because have added these 2 options to option group as @maobaolong suggested, if there are no these  2 options, it would not run into this run method. Do we need to add else-clause to catch the illegal argument?

Reply:
> ... Do we need to add else-clause to catch the illegal argument?

Please add it.  It is similar to a precondition.  Adding preconditions in general is a good practice.

Reply:
Got it, thank you!